### PR TITLE
:memo: Clarify mount troubleshooting

### LINKED ISSUE #1992: Update mount troubleshooting
### Where on docs.platform.sh should be changed?

 https://docs.platform.sh/configuration/app/troubleshoot-mounts.html

### What exactly should be updated?

Take the information from questions and other sections on errors and turn it into a "Troubleshoot mounts" page. Leave the reference information as is for a future reference page.

Update for:

* Content accuracy
* Includes what it should
* Doesn't include what doesn't fit
* Fits template
* Fits style guide

The section on public and private uploads shouldn't be specific to PHP

### LINKED ISSUE #2144: Clarify mounts at build vs. runtime
### Where on docs.platform.sh should be changed?

https://docs.platform.sh/configuration/app/app-reference.html#mounts

### What exactly should be updated?

Users were uncertain whether files in a mount were available during the build or only during runtime. Clarify that at runtime the mount will shadow the listed folders, so adding files during the build is inadvisable. Make sure it's clear what files are available when.

### Additional context

See project for additional context

### PR DESCRIPTION
**Pull request by CollierCZ**
## Why

Closes #1992
Closes #2144

## What's changed

- Changed troubleshooting items into template formate
- Added one item on warning about overlapping directories
- Removed items not related to troubleshooting and not recommended
- Added clarity on when mounts available

### CODE REVIEW

File: docs/src/configuration/app/troubleshoot-mounts.md [hunk 1]
```diff
@@ -58,43 +86,23 @@ mounts:
...
 
-In this configuration, there are two mount points as seen from the application: `~/private` and `~/secret`.
-However, the `secret` mount points to a directory that's also under the mount point for `private`.
-So the `secret` path and the `private/secret` path is the exact same directory.
-
-Although this configuration doesn't cause any technical issues, it may be quite confusing so is generally not recommended.
-
-## Check the size of mounts
-
-You can use standard commands such as `df -ah` to find the total disk usage of mounts
```
Comment (trolologuy) [inline docs/src/configuration/app/troubleshoot-mounts.md:84]: :pick: Why did we remove the `df` and `du` commands?
Indeed the CLI is enough for most use cases, but df and the likes are useful for more advanced troubleshooting.

**Review by trolologuy — APPROVED** [review]
- [x] Final built pages are working as expected.
- [x] The information is correct.
- [x] The information is comprehensive.
- [x] The information is concise.
- [x] The information is clear.
- [x] The changes follow the content and markup style guides.

File: docs/src/configuration/app/troubleshoot-mounts.md (hunk 1 above)
Comment (CollierCZ) [reply docs/src/configuration/app/troubleshoot-mounts.md:84]: The CLI uses those commands behind the scenes and it didn't seem to be important to expose that.

Also, if they are "standard commands", then I'm not sure they need to be documented in our docs. I feel like we should focus mostly on documenting what is unique to Platform.

We can add them back if there are specific scenarios they help with, but they just seemed to be a less transparent way of accomplishing the same thing. So I stuck with the one command to simplify it.

**Comment by CollierCZ** [comment]
Rerunning the script made that be fine.

**Pull request merged**